src/App.tsx

```diff
@@ -1,27 +1,61 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [buttonClicked, setButtonClicked] = useState('');
+  const allButton = 'all-button';
+  const fiveButton = 'first-five-button';
+  const redButton = 'red-button';
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    switch (buttonClicked) {
+      case allButton:
+        getAll().then(setGoods);
+        break;
+      case fiveButton:
+        get5First().then(setGoods);
+        break;
+      case redButton:
+        getRedGoods().then(setGoods);
+        break;
+      default:
+    }
+  }, [buttonClicked]);
```

Comment:
Better create 3 differrenr handlers instead of state + useEffect - it will be more clear